src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1038,7 +1038,7 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
 
         BuildData previousBuildData = getBuildData(build.getPreviousBuild());   // read only
         BuildData buildData = copyBuildData(build.getPreviousBuild());
-        build.addAction(buildData);
```

Comment:
I brainfarted there and just looked at `Actionable` itself. Durr. I'd still prefer to avoid adding it rather than remove it - better user experience to not have a `BuildData` there briefly and then vanishing, IMO.